docs/tutorial/source-plugin-tutorial.md

```diff
@@ -0,0 +1,975 @@
+
+Now when you run your site, you will also be able to query a `childImageSharp` field on the `post.remoteImage`:
+
+```graphql
+query {
+  allPost {
+    nodes {
+      remoteImage {
+        // highlight-start
+        childImageSharp {
+          id
+        }
+        // highlight-end
+      }
+    }
+  }
+}
+```
+
+With data available, you can now query optimized images to use with the `gatsby-image` component in a site! You will need to install `gatsby-image` before you can use it.
+
+### Create foreign key relationships between data
+
+To link the posts to the authors, Gatsby needs to be aware that the two are associated, and how. You have already implemented one example of this when Gatsby inferred a connection between a `remoteImage` and the remote file from Unsplash.
+
+The best approach for connecting related data is through customizing the GraphQL schema. By implementing the `createSchemaCustomization` API, you can specify what data and its types can exist on nodes in Gatsby's GraphQL data layer.
+
+Copy this code and add it to the `source-plugin` in the `gatsby-node.js` file:
+
+```javascript:title=source-plugin/gatsby-node.js
+exports.createSchemaCustomization = ({ actions }) => {
+  const { createTypes } = actions
+  createTypes(`
+    type Post implements Node {
+      id: ID!
+      slug: String!
+      description: String!
+      imgUrl: String!
+      imgAlt: String!
+      # create relationships between Post and File nodes for optimized images
+      remoteImage: File @link
+      # create relationships between Post and Author nodes
+      author: Author @link(from: "author.name" by: "name")
+    }
+    type Author implements Node {
+      id: ID!
+      name: String!
+    }`)
+}
+```
+
+The `author: Author @link(from: "author.name" by: "name")` line tells Gatsby to look for the value on the `Post` node at `post.author.name` and relate it with an `Author` node with a matching `name`. This demonstrates the ability to link using more than just an ID.
+
+The line `remoteImage: File @link` tells Gatsby to look for a `remoteImage` field on a `Post` node and link it to a `File` node with the ID there.
+
+Now, instead of using inference in for the `remoteImage` field, you can take off the `___NODE` suffix. You can update the code in the `onCreateNode` API now like this:
+
+```diff:title=source-plugin/gatsby-node.js
+  if (fileNode) {
+-   node.remoteImage___NODE = fileNode.id
++   node.remoteImage = fileNode.id
+  }
+```
+
+Now running the site will allow you to query authors and remoteImages from the post nodes!
+
+```graphql
+query {
+  allPost {
+    nodes {
+      id
+      // highlight-start
+      author {
+        name
+      }
+      remoteImage {
+        id
+      }
+      // highlight-end
+    }
+  }
+}
+```
```

Comment:
Might be worth pointing out that forgetting to install `gatsby-image` will result in `remoteImage` being null at this point. Ask me how I know. 😅